src/main/cpp/src/utils/vid_mapper_sql.cc

```diff
@@ -0,0 +1,292 @@
+                                       contig_name,
+                                       length,
+                                       tiledb_column_offset);
+    m_contig_begin_2_idx[contig_idx].first = tiledb_column_offset;
+    m_contig_begin_2_idx[contig_idx].second = contig_idx;
+    m_contig_end_2_idx[contig_idx].first = (tiledb_column_offset + length - 1); 
+    m_contig_end_2_idx[contig_idx].second = contig_idx;
+  }
+
+  dbi_result_free(result);
+  return(GENOMICSDB_VID_MAPPER_SUCCESS);
+}
+
+int SQLBasedVidMapper::load_callset_info() {
+  std::stringstream ss;
+  ss <<"select a.id, a.name from callset a, ";
+  ss <<"db_array b, workspace c, callset_to_db_array_association d ";
+  ss <<"where ((c.name='" <<m_work_space <<"') and (c.id = b.workspace_id) ";
+  ss <<"and (b.name = '" <<m_array_name <<"') and ";
+  ss <<"(b.id = d.db_array_id) and (d.callset_id = a.id))";
+
+  std::string query = ss.str();
+  dbi_result result = dbi_conn_query(m_conn, query.c_str());
+  if (NULL == result) {
+    const char* errmsg = NULL;
+    dbi_conn_error(m_conn, &errmsg);
+    std::string error_message = (DBQUERY_FAILED + " : " + errmsg);
+    THROW_EXCEPTION(error_message);
+  }
+
+  int num_callsets = (int) dbi_result_get_numrows(result);
+  m_row_idx_to_info.resize(num_callsets);
+  m_max_callset_row_idx = -1;
+
+  while (dbi_result_next_row(result)) {
+    std::string callset_name = dbi_result_get_string(result, DBTABLE_CALLSET_COLUMN_NAME.c_str());
+    int64_t row_idx = dbi_result_get_longlong(result, DBTABLE_CALLSET_COLUMN_ID.c_str());
+    int64_t file_idx = row_idx;
+    m_max_callset_row_idx = std::max(m_max_callset_row_idx, row_idx);
+    m_callset_name_to_row_idx[callset_name] = row_idx;
+    m_row_idx_to_info[row_idx].set_info(row_idx, callset_name, file_idx, 0);
+  }
+
+  dbi_result_free(result);
+  return(GENOMICSDB_VID_MAPPER_SUCCESS);
+}
+
+int SQLBasedVidMapper::load_field_info() {
+  std::stringstream ss;
+  ss <<"select a.id, a.field, a.type, a.is_filter, a.is_format, a.is_info, ";
+  ss <<"a.length_type, a.length_intval, a.field_combine_op ";
+  ss <<"from field a, db_array b, workspace c ";
+  ss <<"where ((c.name='" <<m_work_space <<"') and (c.id = b.workspace_id) ";
+  ss <<"and (b.name = '" <<m_array_name <<"') and ";
+  ss <<"(a.field_set_id = b.field_set_id)) ";
+  ss <<"order by a.id";
+
+  std::string query = ss.str();
+  dbi_result result = dbi_conn_query(m_conn, query.c_str());
+  if (NULL == result) {
+    const char* errmsg = NULL;
+    dbi_conn_error(m_conn, &errmsg);
+    std::string error_message = (DBQUERY_FAILED + " : " + errmsg);
+    THROW_EXCEPTION(error_message);
+  }
+
+  int num_fields = (int) dbi_result_get_numrows(result);
+  m_field_name_to_idx.reserve(num_fields);
+  m_field_idx_to_info.resize(num_fields);
+
+  while (dbi_result_next_row(result)) {
+    int64_t field_id = dbi_result_get_longlong(result, DBTABLE_FIELD_COLUMN_ID.c_str());
+    int64_t field_idx = (field_id - 1);
```

Comment:
_field_id_ returned from the DB could be much larger than the _num_fields_ for this specific array when the mapping DB has information about multiple arrays. Computing _field_idx_ from the _field_id_ will cause writes to un-allocated memory in the lines below. Use an incrementing counter for _field_idx_.